### PR TITLE
CORE-6384 add s3 publishing to repo

### PR DESCRIPTION
**Pull request by ronanbrowne**
Leaving in draft while finalize what submodules needed for DP2

For DP2 QA testing we are are adding the ability to publish to a isolated staging area (S3 bucket which can act as a maven repo).

Logic will be triggered from the private shared library repo where needed details will be passed.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -242,6 +242,19 @@ configure(publishProjects) { Project subproject ->
                 }
             }
         }
+
+        if (project.hasProperty('maven.repo.s3') && subproject.name.equals("cordapp-cpk2")){
```
Comment (chrisr3) [inline build.gradle:246]: So is this going to publish `cordapp-cpk2` to S3 while still publishing all of the other plugins to Artifactory? Perhaps you want:
```gradle
def publishProjects = hasProperty('maven.repo.s3') ? [ project(':cordapp-cpk2') ] : childProjects.values()
```
and _then_ define the `repositories` block.

**Review by chrisr3 — COMMENTED** [review]
That `if` statement could grow to become rather ugly.

File: build.gradle (hunk 1 above)
Comment (ronanbrowne) [reply build.gradle]: Thanks @chrisr3 updated

File: build.gradle [hunk 2]
```diff
@@ -241,6 +242,18 @@ configure(publishProjects) { Project subproject ->
                     }
                 }
             }
+
+
+        }
```
Comment (chrisr3) [inline build.gradle:247]: whitespace error ... :wink:

File: build.gradle [hunk 3]
```diff
@@ -241,6 +242,18 @@ configure(publishProjects) { Project subproject ->
...
+        repositories {
+            maven {
+                url = project.findProperty('maven.repo.s3')
+                credentials(AwsCredentials) {
+                    accessKey "${System.getenv('AWS_ACCESS_KEY_ID')}"
+                    secretKey "${System.getenv('AWS_SECRET_ACCESS_KEY')}"
+                    sessionToken "${System.getenv('AWS_SESSION_TOKEN')}"
+                }
+            }
         }
```
Comment (chrisr3) [inline build.gradle:257]: But what if `maven.repo.s3` is undefined? Doesn't this break publishing just to Artifactory? I was expecting:
```gradle
if (hasProperty('maven.repo.s3')) {
    repositories {
        maven {
            url = getProperty('maven.repo.s3')
            ...
        }
    }
}
```

I believe you can also assign a name to this repository:
```gradle
maven {
    name = 'AWS'
    url = ...
}
```

**Review by chrisr3 — COMMENTED** [review]
Just wonderin'...

File: build.gradle [hunk 4]
```diff
@@ -242,6 +243,19 @@ configure(publishProjects) { Project subproject ->
                 }
             }
         }
+        if (subproject.hasProperty('maven.repo.s3')) {
```
Comment (chrisr3) [inline build.gradle:246]: I suspect the `maven.repo.s3` property is _global_, but whatever :grin:.

**Review by chrisr3 — APPROVED** [review]
:sparkles: Good enough :sparkles:

:wink:

**Pull request merged**